packages/vue-server-renderer/types/index.d.ts

```diff
@@ -0,0 +1,44 @@
+import Vue = require('vue');
+import { Readable } from 'stream';
+
+export declare function createRenderer(options?: RendererOptions): Renderer;
+
+export declare function createBundleRenderer(bundle: any, options?: BundleRendererOptions): BundleRenderer;
+
+type RenderCallback = (err: Error, html: string) => void;
+
+interface Renderer {
+  renderToString(vm: Vue, callback: RenderCallback): void;
+  renderToString(vm: Vue, context: object, callback: RenderCallback): void;
+
+  renderToStream(vm: Vue, context?: object): Readable;
+}
+
+interface BundleRenderer {
+  renderToString(callback: RenderCallback): void;
+  renderToString(context: object, callback: RenderCallback): void;
+
+  renderToStream(context?: object): Readable;
+}
+
+interface RendererOptions {
+  template?: string;
+  inject?: boolean;
+  shouldPreload?: (file: string, type: string) => boolean;
+  cache?: RenderCache;
+  directives?: {
+    [key: string]: Vue.DirectiveOptions | Vue.DirectiveFunction
```

Comment:
Note server side directive will have different implementation. http://ssr.vuejs.org/en/api.html#directives
https://github.com/vuejs/vue/blob/d333a498e93acf2d2844fd396e8544658958a3c6/src/server/render.js#L308

IMHO direcitives will be 

```ts
 [key: string]: (vnode: VNode, dir: VNodeDirective) => void
```